crates/rustc_codegen_spirv/src/builder/load_store.rs

```diff
@@ -0,0 +1,506 @@
+
+                    composite_components.push(
+                        self.recurse_adt_for_loads(
+                            uint_ty,
+                            uniform_uint_ptr,
+                            bindless_idx,
+                            base_offset_var,
+                            element_offset_literal + offset,
+                            *ty,
+                            sets,
+                        )
+                        .def(self),
+                    );
+                }
+
+                let adt = data.def(rustc_span::DUMMY_SP, self);
+
+                self.emit()
+                    .composite_construct(adt, None, composite_components)
+                    .unwrap()
+                    .with_type(adt)
+            }
+            SpirvType::Vector { count, element } => {
+                let mut composite_components = vec![];
+
+                for offset in 0..count {
+                    composite_components.push(
+                        self.recurse_adt_for_loads(
+                            uint_ty,
+                            uniform_uint_ptr,
+                            bindless_idx,
+                            base_offset_var,
+                            element_offset_literal + offset,
+                            element,
+                            sets,
+                        )
+                        .def(self),
+                    );
+                }
+
+                let adt = data.def(rustc_span::DUMMY_SP, self);
+
+                self.emit()
+                    .composite_construct(adt, None, composite_components)
+                    .unwrap()
+                    .with_type(adt)
+            }
+            SpirvType::Float(bits) => {
+                let loaded_as_int = self
+                    .load_from_u32(
+                        bits,
+                        false,
+                        uint_ty,
+                        uint_ty,
+                        uniform_uint_ptr,
+                        bindless_idx,
+                        base_offset_var,
+                        element_offset_literal,
+                        sets,
+                    )
+                    .def(self);
+
+                self.emit()
+                    .bitcast(result_type, None, loaded_as_int)
+                    .unwrap()
+                    .with_type(result_type)
+            }
+            SpirvType::Integer(bits, signed) => self.load_from_u32(
+                bits,
+                signed,
+                result_type,
+                uint_ty,
+                uniform_uint_ptr,
+                bindless_idx,
+                base_offset_var,
+                element_offset_literal,
+                sets,
+            ),
+            _ => {
+                self.fatal(&format!(
+                    "Unhandled case for `internal_buffer_load` return / args: {:?}",
+                    data
+                ));
+            }
```

Comment:
Same here, should support more types.

Reply:
I've added error message for un-supported types, and added support for fixed size arrays.